bftengine/src/bftengine/ReplicaConfigSingleton.cpp

```diff
@@ -0,0 +1,141 @@
+//Concord
+//
+//Copyright (c) 2019 VMware, Inc. All Rights Reserved.
+//
+//This product is licensed to you under the Apache 2.0 license (the "License").  You may not use this product except in compliance with the Apache 2.0 License.
+//
+//This product may include a number of subcomponents with separate copyright notices and license terms. Your use of these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE file.
+
+#include "ReplicaConfigSingleton.hpp"
+
+#include "ReplicaConfig.hpp"
+
+class IThresholdSigner;
+class IThresholdVerifier;
+
+namespace bftEngine
+{
+
+ReplicaConfigSingleton::ReplicaConfigSingleton() : is_initialized_(false) {}
+
+ReplicaConfigSingleton& ReplicaConfigSingleton::GetInstance(const ReplicaConfig* config)
```

Comment:
Could you please add a comment, either here or in the header file, noting that this function is not thread-safe, so it is the caller's job to ensure that only a single thread calls this with a non-null pointer at a time?

Reply:
Thanks for the review!

Reply:
I ran `make test` and the simpleTest as Andrew pointed out below.